### PR TITLE
User auth and comments backend

### PR DESCRIPTION
**Pull request by luschnat-ziegler**
Adds endpoints to back-end. Everything (seems to) work(s).

### CODE REVIEW

File: backend/src/Repository/TokenRepository.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, Token::class);
+    }
+
+    public function create(Object $user): Token
+    {
+        $validUntil = new \Datetime();
+        $validUntil->modify('+1 day');
```
Comment (FabianAlleblas) [inline backend/src/Repository/TokenRepository.php:25]: `new \Daytime('+ 1 day')`

also works

File: backend/src/Controller/UserController.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    public function register(
+        Request $request,
+        SerializerInterface $serializer,
+        UserRepository $repository
+    ): JsonResponse
+    {
+        $user = $serializer->deserialize($request->getContent(), User::class, 'json');
+        
+        $emailInUse = $repository->findBy(['email' => $user->getEmail()]);
```
Comment (FabianAlleblas) [inline backend/src/Controller/UserController.php:58]: 👍

**Review by FabianAlleblas — COMMENTED** [review]
Looks good to me 🙌 , you could already add a function that deletes your unvalid token to the validateUser process.

File: backend/src/Repository/TokenRepository.php (hunk 1 above)
Comment (luschnat-ziegler) [reply backend/src/Repository/TokenRepository.php:25]: And is nicer, thank you!

**Pull request merged**